test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -277,11 +275,33 @@ func (sm *StackManager) deleteStack(ctx context.Context, stackID string, stack *
 		logger = logWriter
 	}
 	progressStreamsDestroyOption := sm.getProgressStreamsOnDestroy(logger)
+	//initialize datadog event sender
+	if ddEventSender == nil {
+		ddEventSender = newDatadogEventSender(logger)
+	}
 
-	_, err = stack.Destroy(destroyContext, progressStreamsDestroyOption, optdestroy.DebugLogging(loggingOptions))
-	cancel()
-	if err != nil {
-		return err
+	downCount := 0
+	var destroyErr error
+	for {
+		downCount++
+		destroyContext, cancel := context.WithTimeout(ctx, stackDestroyTimeout)
+		_, destroyErr = stack.Destroy(destroyContext, progressStreamsDestroyOption, optdestroy.DebugLogging(loggingOptions))
+		cancel()
+		if err == nil {
+			sendEventToDatadog(ddEventSender, fmt.Sprintf("[E2E] Stack %s : success on Pulumi stack destroy", stackID), "", []string{"operation:destroy", "result:ok", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", downCount)})
+			break
+		}
+		sendEventToDatadog(ddEventSender, fmt.Sprintf("[E2E] Stack %s : error on Pulumi stack destroy", stackID), destroyErr.Error(), []string{"operation:destroy", "result:fail", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", downCount)})
+
+		if downCount > stackUpMaxRetry {
+			fmt.Printf("Giving up on error during stack destroy: %v\n", destroyErr)
+			break
```

Comment:
nit - you could return `destroyErr` directly here and remove the if statement after the for loop

Reply:
Indeed, thanks!